### ISSUE TITLE
Share file caching across markbind instances

### ISSUE THREAD

**Issue by nicholaschuayunzhi** [MarkBind/markbind#138]
https://github.com/MarkBind/markbind/blob/eed11dd1169acb2f08d63ab08da5544bb2571156/lib/parser.js#L73-L76

Currently each `Parser` instance has its own cache. Each `Page` creates its own `parser` or `markbinder` instance and thus the cache is not shared. Considering there is a lot file reuse in `Markbind`, I feel we should have a common cache to cut down on the IO operations.

I propose having a static instance of file cache for `Parser` class. Possibly can abstract as its own module?

# Things to consider
Not reading from cache when file is changed during live reload.

Related #88

**Comment by nicholaschuayunzhi** [comment]
## Update

### Before (Build time, number of cache hits)
- build: 34344.112ms, 5143
- build: 36372.435ms, 5143
- build: 34837.442ms, 5143

### After
- build: 33073.394ms, 13563
- build: 26338.988ms, 13563
- build: 29040.912ms, 13563

https://github.com/nicholaschuayunzhi/markbind/tree/138-static-file-cache
https://github.com/nicholaschuayunzhi/markbind-cli/tree/138-static-file-cache

Waiting for consensus for #139

**Comment by damithc** [comment]
@nicholaschuayunzhi can we resurrect this PR?. Any performance gain in the build is a worthy cause.

**Comment by nicholaschuayunzhi** [comment]
Sorry for the late response, I've been trying this out for a bit.

While there is an increase in Cache Hits, there does not seem to be improvements in memory and performance. I'm not too sure of the reasons why - perhaps IO operations is not bottleneck of the build process? Tested on 2103 website-base

`Before`
```
heapUsed 151.83 MB
Cache Hit: 12510
Cache Miss: 10793
Build Time: 100524.524ms
```
`After`
```
heapUsed 156.57 MB
Cache Hit: 21823
Cache Miss: 1480
Build Time: 114969.759ms
```
I've made the changes here: https://github.com/nicholaschuayunzhi/markbind/commit/83aa07dd57b3e15d681ba31b29d335f9bb69f632
